pkg/apis/vsphereproviderconfig/v1alpha1/vspheremachineproviderstatus_types.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VsphereMachineProviderStatus defines the observed state of VsphereMachineProviderConfig
+// +k8s:openapi-gen=true
+type VsphereMachineProviderStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	TaskRef string `json:"taskRef"`
```

Comment:
I don't think this TaskRef works, or at least not in the way it is currently being used. 

When trying to patch in the TaskRef you get a `422 Unprocessable Entity` error, previously TaskRef was being set by Machine.UpdateStatus()
Not sure if this is due to how the API is defined (I tried making it optional with no difference) trying to patch a `runtime.RawExtension` in a status

Reply:
Thank you @moshloop for helping fix the broken task logic!

Reply:
I am testing against ESXi, so maybe it is a new code path I have introduced - will submit a PR and see if that passes on VMC